tests/integration/test___init__run_conflict.py

```diff
@@ -0,0 +1,187 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Integration tests for running the action where there is a merge conflict."""
+
+# This test is fairly complex as it simulates sequential action runs
+# pylint: disable=too-many-arguments,too-many-locals,too-many-statements
+# The tests for reconcile are similar, although it is better to have some minor duplication than
+# make the tests less clear
+# pylint: disable = duplicate-code
+
+import logging
+from base64 import b64encode
+from itertools import chain
+from pathlib import Path
+from unittest.mock import MagicMock
+from urllib.parse import urlparse
+
+import pytest
+from github.ContentFile import ContentFile
+
+from src import constants, exceptions, metadata, run
+from src.discourse import Discourse
+
+from .. import factories
+from ..unit.helpers import assert_substrings_in_string, create_metadata_yaml
+
+pytestmark = pytest.mark.conflict
+
+
+@pytest.mark.asyncio
+@pytest.mark.usefixtures("patch_create_repository_client")
+async def test_run_conflict(
```

Comment:
Shouldn't we apply the same test but in dry-run ?

Reply:
Yes good point, I'll add a stage in dry-run mode